python/ray/data/tests/test_dynamic_block_split.py

```diff
@@ -3,67 +3,268 @@
 import pytest
 
 import ray
+from ray.data._internal.lazy_block_list import LazyBlockList
 from ray.data.block import BlockMetadata
 from ray.data.datasource import Datasource
 from ray.data.datasource.datasource import ReadTask, Reader
 
 from ray.tests.conftest import *  # noqa
 
 
-def test_read_large_data(ray_start_cluster):
-    # Test 20G input with single task
-    num_batch = 20
+# Data source generates random bytes data
+class RandomBytesDatasource(Datasource):
+    def create_reader(self, **read_args):
+        return RandomBytesReader(read_args["num_batches"], read_args["batch_size"])
+
+
+class RandomBytesReader(Reader):
+    def __init__(self, num_batches: int, batch_size: int):
```

Comment:
Can we use "block" instead of "batch" here? It has block concept at datasource but not batch yet.